api/nodes/serializers.py

```diff
@@ -459,6 +473,18 @@ def get_registration_links_count(self, obj):
                 count += 1
         return count
 
+    def get_linked_by_nodes_count(self, obj):
+        return NodeRelation.objects.filter(child=obj, is_node_link=True).select_related('parent').exclude(parent__type='osf.collection').exclude(parent__type='osf.registration').count()
+
+    def get_linked_by_registrations_count(self, obj):
+        return NodeRelation.objects.filter(child=obj, is_node_link=True, parent__type='osf.registration').select_related('parent').exclude(parent__type='osf.collection').count()
```

Comment:
Same question here: Do you need to filter out withdrawn registrations linking to here, or is that handled elsewhere?

Reply:
Also: I don't think `select_related('parent')` is necessary here, since you're already doing the JOIN (`parent__type`) and you're only getting a count (not accessing `node_relation.parent`). In fact, select_related might actually be slowing this down because it's fetching more than it needs.